tycho-its/src/test/java/org/eclipse/tycho/test/util/HttpServer.java

```diff
@@ -153,26 +169,47 @@ public void stop() throws Exception {
 
 	public String addServer(String contextName, final File content) {
 		ServletContextHandler context = new ServletContextHandler(contexts, URIUtil.SLASH + contextName);
+		context.setInitParameter("org.eclipse.jetty.servlet.Default.dirAllowed", "false");
 		context.setResourceBase(content.getAbsolutePath());
+		context.addServlet(DefaultServlet.class, URIUtil.SLASH);
+		registerContext(context);
+		return getUrl(contextName);
+	}
 
-		MonitoringServlet monitoringServlet = new MonitoringServlet();
-		contextName2servletsMap.put(contextName, monitoringServlet);
-		context.addServlet(new ServletHolder(monitoringServlet), URIUtil.SLASH);
-		contexts.addHandler(context);
-		try {
-			context.start();
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
+	/**
+	 * 
+	 * @param contextName         - a path prefix to handle
+	 * @param relativeUrlToNewUrl - redirecting function. Takes a path within
+	 *                            context starting with slash and return an new
+	 *                            absolute URL to redirect to (Location header
+	 *                            value).
+	 * @return an URL prefix to redirect from
+	 */
+	public String addRedirect(String contextName, Function<String, String> relativeUrlToNewUrl) {
+		ServletContextHandler context = new ServletContextHandler(contexts, URIUtil.SLASH + contextName);
+		context.addServlet(new ServletHolder(new RedirectServlet(relativeUrlToNewUrl)), URIUtil.SLASH);
+		registerContext(context);
 		return getUrl(contextName);
+
 	}
 
 	public String getUrl(String contextName) {
 		return "http://localhost:" + port + "/" + contextName;
 	}
 
 	public List<String> getAccessedUrls(String contextName) {
-		return contextName2servletsMap.get(contextName).getAccessedURIs();
+		synchronized (contextName2accessedUrls) {
+			return List.copyOf(contextName2accessedUrls.get(contextName));
```

Comment:
This seem to introduce test failures now as some test afterwards modify the collection see:
https://ci.eclipse.org/tycho/job/tycho-github/job/PR-2945/1/

Reply:
This was a bit of a struggle, as it is not trivial to run integration tests locally for Tycho 5.0.x.
https://github.com/eclipse-tycho/tycho/issues/2946
https://github.com/eclipse-tycho/tycho/issues/2947